### PR TITLE
Sprint 36 TLT-2160 Use correct display name for sender

### PR DESCRIPTION
**Pull request by rascalking**
Mailgun seems to only use the email address for the `sender` field in their POST, so looking for its display name isn't ever going to work.  Instead, assuming the `sender` and `from` fields agree on the email address, let's use the `from` display name.  If that's missing, then fall back on enrollee lookup.

### CODE REVIEW

**Comment by elliottyates** [comment]
@elliottyates

File: mailgun/tests.py [hunk 1]
```diff
@@ -261,6 +265,133 @@ def test_bounce_loop_detection(self, mock_ml_get, mock_ci_get, mock_ss_filter, m
...
+    @patch('mailgun.route_handlers.get_name_for_email')
+    @patch('mailgun.route_handlers.SuperSender.objects.filter')
+    @patch('mailgun.route_handlers.CourseInstance.objects.get_primary_course_by_canvas_course_id')
+    @patch('mailgun.route_handlers.MailingList.objects.get_or_create_or_delete_mailing_list_by_address')
+    def test_sender_display_name_from_field(self, mock_ml_get, mock_ci_get, mock_ss_filter,
+                                            mock_get_name_for_email):
+        '''
+        TLT-2160
+        Verifies that if the mailgun `sender` field lacks a display name, we'll
+        use the one in the mailfun `from` field before resorting to enrollee
```
Comment (elliottyates) [inline mailgun/tests.py]: 'mailfun' -- Freudian slip?

**Comment by elliottyates** [comment]
Looks good. :+1: Does the parsing library handle left/right-whitespace padding? If not, do we want to be paranoid and `strip()` the display name strings before checking for `None`? Or just let their display name indeed be possibly blank if they've configured their client that way?

File: mailgun/tests.py (hunk 1 above)
Comment (rascalking) [inline mailgun/tests.py]: I think a Freudian slip would not have substituted `fun`.  :)

**Comment by rascalking** [comment]
Yeah, it's my understanding that they handle excess leading/trailing whitespace in the display name.

Hah, forgot they provide a grammar for their parser:
https://github.com/mailgun/flanker/blob/master/docs/User%20Manual.md#grammar

**Pull request merged**